### ISSUE TITLE
Random freezing of Godot IDE on Linux. Happens enough times to be disconcerting.  Anyway to determine cause?

### ISSUE THREAD

**Issue by wyattbiker** [godotengine/godot#95316]
### Tested versions

v4.3.rc2.official [3978628c6]

### System information

Linux Kubuntu 23.04 KDE Plasma Version: 5.27.4 KDE Frameworks Version: 5.104.0 Qt Version: 5.15.8

### Issue description

I am using Linux Kubuntu KDE.

When using the Godot IDE sometimes it freezes or acts erratic/nonsensical as I am editing. I found I can fix this by minimizing Godot  (or clicking away to another application) and then maximize with no losses. I am not sure keyboard still works, as usually my doing something in script or inspector editors.

I am so used to doing this, but I am wondering if anyone else has a similar issue and is there something I should look for when it happens again. 

This has been happening in the past few versions of 4.3. I don't know if it happened in 4.2. I cannot pin point it but it happens frequently enough to be annoying.

### Steps to reproduce

see above

### Minimal reproduction project (MRP)

see above

**Comment by huwpascoe** [comment]
> When using the Godot IDE sometimes it freezes or acts erratic/nonsensical as I am editing

A freeze, could be anything. But the erratic/nonsense, that could give us a lead. Please describe this strange behavior in as much detail as possible.

**Comment by akien-mga** [comment]
I can't reproduce it on Fedora 40 with KDE Plasma 6.1.3 on Wayland.

Since you're using Ubuntu 23.04 which is EOL since January 2024, I would encourage you to upgrade to 24.04 LTS if you can, which _may_ fix this issue, if it was caused by a bug in the GPU drivers, KWin, or other parts of the Linux stack.

**Comment by davidpkj** [comment]
I may have a similar issue on Arch Linux with Xmonad running Godot v4.3.stable.arch_linux. I had this issue with 4.2 as well, but can't say if it was there before.

When I switch to a different workspace for some time (usually at least 5-30s) and then switch back, the IDE seems to not draw anything for a few seconds before resuming as if nothing happened.

So no weird behaviour, except that it is literally unusable while it "recovers". During this "recovery" the place were the window would be is filled with whatever was in its place before I switched workpsaces. So it seems like Godot freezes. But this only happens when having a (any) project open. In the launcher menu thing this issue does not occur. Running with `--verbose` did not log anything right before, during or after the freeze.

**Comment by Calinou** [comment]
> When I switch to a different workspace for some time (usually at least 5-30s) and then switch back, the IDE seems to not draw anything for a few seconds before resuming as if nothing happened.

Can you reproduce this with **Update Continuously** enabled in the Editor Settings?

**Comment by davidpkj** [comment]
No, that fixes my issue, thanks. How did I miss that setting? :thinking:

**Comment by wyattbiker** [comment]
I don't have mine set either, though my issue is pretty random and as soon as switch to another app and back fixes itself. Ill test that setting anyway it but it does say it greatly increases CPU usage. 

> If true, redraws the editor every frame even if nothing has changed on screen. When this setting is enabled, the update spinner displays in red (see Interface > Editor > Show Update Spinner).
> 
> Warning: This greatly increases CPU and GPU utilization, leading to increased power usage. This should only be enabled for troubleshooting purposes.

**Comment by SubhadeepJasu** [comment]
Can confirm this happens in Windows as well. The editor freezes indefinitely when saving, or switching to another scene in a tab, etc. I am having to constantly restart the editor to keep working on my game. The update continuously flag doesn't fix this. There are no error messages. This has slowed down development significantly and is now a major blocker.

**Comment by Calinou** [comment]
@wyattbiker @SubhadeepJasu Can you reproduce this with **Single Window Mode** enabled in the Editor Settings

**Comment by costas1000** [comment]
@Calinou I am now using v4.3.stable.official [77dcf97d8]. I've set it to Single Window Mode. I am also upgraded to 

Operating System: Kubuntu 24.04
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.13
Kernel Version: 6.8.0-51-generic (64-bit)
Graphics Platform: X11


As I said, it is pretty random. I will keep the Single Window Mode check for a while. Ill post if it happens again.

**Comment by wyattbiker** [comment]
@Calinou Switched to Single Window Mode. No change, will keep it checked for a while. If it happens will let you know.

Current Setup:
v4.3.stable.official [77dcf97d8]

Operating System: Kubuntu 24.04
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.13
Kernel Version: 6.8.0-51-generic (64-bit)
Graphics Platform: X11

**Comment by DanikVitek** [comment]
I experience the similar freezing issues. I've determined, that the CPU spikes to 100% every time it lags, and the whole system freezes. Also, for some reason, this happens even when opening some other apps (e.g. Ghostty), while the Godot runs.

Godot: v4.4.1.stable.mono.official [49a5bc7b6]
OS: Linux Mint 22.1 Cinnamon 6.4.8
Kernel: 6.8.0-64-generic
CPU: AMD Ryzen 5 5600H with Radeon Graphics × 6
GPU: GeForce RTX 3050 Mobile
Win interface: X11

[system-info.txt](https://github.com/user-attachments/files/21304378/system-info.txt)

Here's the video of the behavior:

https://github.com/user-attachments/assets/1947532e-3f88-4540-a177-439c153f5645
